### ISSUE TITLE
`LET`/`SET` values cannot be used in parallel commands such as `BUILD`

### ISSUE THREAD

**Issue by idodod** [earthly/earthly#3997]
**What went wrong?**

Trying to defined the name of the target in a `LET` statement, the name of the target is not properly expanded and the target cannot be built.

```Earthfile
VERSION 0.8

FROM alpine

foo:
    LET target=+bar
    BUILD $target

bar:
    RUN echo hello from bar
```

`earthly +foo`

**What should have happened?**

The command should succeed in eventually building `+bar`.
chan

**Other Helpful Information**

Changing the `LET` command to an `ARG` allows the target to be built.

It appears LET/SET do not block parallel commands, thus BUILD is executed before LET is evaluated.

**Comment by idodod** [comment]
This also fixes a bug where a BUILD might get executed twice with different values of an arg.
Consider the folliwing:
```Earthfile

FROM alpine

foo:
   LET my_val=abc
   BUILD +bar --my_arg=$my_val
   
bar:
  ARG my_arg="123"
  RUN echo $my_arg   
```

`earthly +foo` would result in +bar getting executed twice - once with my_arg="" and once with my_arg="abc", where the latter is the intended behavior.

**Comment by idodod** [comment]
will be included in next releases (v0.8.8)
